packages/dev/core/src/Shaders/ssao2.fragment.fx

```diff
@@ -113,125 +112,119 @@ float viewZToOrthographicDepth( in float viewZ, in float near, in float far ) {
+		0.09416557240840277,
+		0.08185740093668367,
+		0.06686304882366484,
+		0.05131881937026997,
+		0.03701085986298716,
+		0.025080919943070733,
+		0.01597052517669494,
+		0.05189331191819471 // center
+		);
+#else
+	// The regular (non-bilateral) blur can use a slightly smaller filter kernel
+	// with mostly the same visual result due to not rejecting samples.
+	int offsets = 7;
+	float blurOffsets[7] = float[7](
+		1.4923475363658985,
+		3.482150445593865,
+		5.471968200192311,
+		7.461809225071293,
+		9.451681868040225,
+		11.441594372573679,
+		13.431554851305808
+		//0,
+		);
+	float blurWeights[8] = float[8](
+		0.1155336786276448,
+		0.10438038205425465,
+		0.08694751807157279,
+		0.06677642827586246,
+		0.047284236906781164,
+		0.03086996605725001,
+		0.01858154019942333,
+		0.05925249961442155 // center
+		);
+#endif
 
 	void main()
 	{
-		#if EXPENSIVE
-		float compareDepth = abs(texture2D(depthSampler, vUV).r);
 		float texelsize = 1.0 / outSize;
 		float result = 0.0;
-		float weightSum = 0.0;
-
-		for (int i = 0; i < SAMPLES; ++i)
-		{
-			#ifdef BILATERAL_BLUR_H
-			vec2 direction = vec2(1.0, 0.0);
-			vec2 sampleOffset = vec2(texelsize * samplerOffsets[i], 0.0);
-			#else
-			vec2 direction = vec2(0.0, 1.0);
-			vec2 sampleOffset = vec2(0.0, texelsize * samplerOffsets[i]);
-			#endif
-			vec2 samplePos = vUV + sampleOffset;
 
-			float sampleDepth = abs(texture2D(depthSampler, samplePos).r);
-			float weight = clamp(1.0 / ( 0.003 + abs(compareDepth - sampleDepth)), 0.0, 30000.0);
-
-			result += texture2D(textureSampler, samplePos).r * weight;
-			weightSum += weight;
-		}
-
-		result /= weightSum;
-		gl_FragColor.rgb = vec3(result);
-		gl_FragColor.a = 1.0;
+		#if BLUR_BILATERAL
+			// Sample center separately, avoids unstable result on slanted surfaces
+			float compareDepth = abs(texture2D(depthSampler, vUV).r);
+			float weightSum = 333.3 * blurWeights[offsets];
+			result = texture2D(textureSampler, vUV).r * weightSum;
 		#else
-		vec4 color;
-		#ifdef BILATERAL_BLUR_H
-		vec2 direction = vec2(1.0, 0.0);
-		color = blur13Bilateral(textureSampler, vUV, outSize, direction);
-		#else
-		vec2 direction = vec2(0.0, 1.0);
-		color = blur13Bilateral(textureSampler, vUV, outSize, direction);
+			result = texture2D(textureSampler, vUV).r * blurWeights[offsets];
 		#endif
 
-		gl_FragColor.rgb = vec3(color.r);
-		gl_FragColor.a = 1.0;
+		#ifndef BLUR_BYPASS
+			for (int i = 0; i < offsets; ++i)
+			{
+				#ifdef BLUR_H
+					vec2 samplePos1 = vUV + vec2(texelsize * blurOffsets[i], 0.0);
+					vec2 samplePos2 = vUV + vec2(-texelsize * blurOffsets[i], 0.0);
+				#else
+					vec2 samplePos1 = vUV + vec2(0.0, texelsize * blurOffsets[i]);
+					vec2 samplePos2 = vUV + vec2(0.0, -texelsize * blurOffsets[i]);
+				#endif
+
+				float w = blurWeights[i];
+
+				#if BLUR_BILATERAL
+					float depth = abs(texture2D(depthSampler, samplePos1).r);
+					float weight = clamp(1.0 / ( 0.003 + abs(compareDepth - depth)), 0.0, 333.3) * w;

```

Comment:
That is from the previous code.

But if clamp is removed and the difference in depth is very high... Ah, I see. It can never be bigger than 333.3, only smaller and certainly not negative. 👍